File: Modules/HumanResource/PayrollSettings/PayslipDeductedOneTimeDeductions/PayslipDeductedOneTimeDeductionsForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.PayrollSettings.Forms;

[FormScript("PayrollSettings.PayslipDeductedOneTimeDeductions")]
[BasedOnRow(typeof(PayslipDeductedOneTimeDeductionsRow), CheckNames = true)]
public class PayslipDeductedOneTimeDeductionsForm
{
    public int DeductionId { get; set; }
    public int PayslipId { get; set; }
    public int IsActive { get; set; }
    public DateTime InsertDate { get; set; }
    public DateTime UpdateDate { get; set; }
    public DateTime DeleteDate { get; set; }
    public int InsertUserId { get; set; }
    public int UpdateUserId { get; set; }
    public int DeleteUserId { get; set; }
}

File: Modules/ServerTypes/PayrollSettings/PayslipDeductedOneTimeDeductionsForm.ts
import { IntegerEditor, DateEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface PayslipDeductedOneTimeDeductionsForm {
    DeductionId: IntegerEditor;
    PayslipId: IntegerEditor;
    IsActive: IntegerEditor;
    InsertDate: DateEditor;
    UpdateDate: DateEditor;
    DeleteDate: DateEditor;
    InsertUserId: IntegerEditor;
    UpdateUserId: IntegerEditor;
    DeleteUserId: IntegerEditor;
}

export class PayslipDeductedOneTimeDeductionsForm extends PrefixedContext {
    static formKey = 'PayrollSettings.PayslipDeductedOneTimeDeductions';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!PayslipDeductedOneTimeDeductionsForm.init)  {
            PayslipDeductedOneTimeDeductionsForm.init = true;

            var w0 = IntegerEditor;
            var w1 = DateEditor;

            initFormType(PayslipDeductedOneTimeDeductionsForm, [
                'DeductionId', w0,
                'PayslipId', w0,
                'IsActive', w0,
                'InsertDate', w1,
                'UpdateDate', w1,
                'DeleteDate', w1,
                'InsertUserId', w0,
                'UpdateUserId', w0,
                'DeleteUserId', w0
            ]);
        }
    }
}